scripts/init.js

```diff
@@ -160,13 +189,17 @@ export const onLoadComponents = {
     );
   },
 
-  initBlocks() {
+  async initBlocks() {
     // Keep the page hidden until specific components are initialized to prevent CLS
     component.multiInit(this.lcpBlocks).then(() => {
       window.postMessage({ message: 'raqn:components:loaded' });
       document.body.style.setProperty('display', 'block');
     });
-    component.multiInit(this.lazyBlocks);
+
+    await component.multiInit(this.lazyBlocks);
+    // grids must be initialized sequentially starting from the deepest level.
+    // all the blocks that will be contained by the grids must be already initialized before they are added to the grids.
+    component.multiSequentialInit(this.grids);
```

Comment:
did you check performance for this change?

Reply:
As it is sequential only for grids it doesn't really make a big difference but also depends on the number of grids on the page.

We could improve this by switching the whole component initialization to be done async per section which can also speed up grids initialization because then grids need to wait only for the components in that section not on the whole page.

Plus grouping grids by level and initializing all same level grids async and only the groups will be sequential. Not completely sure but I think this will work and same level grids will not clash with each other when initialized async.